### DISCUSSION TITLE
Initializing variables using other variables or materials

### DISCUSSION THREAD

**Discussion by xueyang94** [idaholab/moose#21626] [Q&A General] [answered]
I have a model with a few nonlinear variables. I want to initialize one of the variables in the IC block using the other variables. I have tried FunctionIC, but seems like it cannot define a function of model variables. I am also wondering how to define the IC of a variable using one of the ParsedMaterial properties in the Material block. Thank you.

**Comment by GiudGiud** [comment, marked as answer]
Hello

You wont be able to use ICs I think.
And since they are non linear you can not use AuxKernels. 

So what I would do would be a MultiApp ran on initial, with a couple transfers from_multiapp to overwrite the starting values of the nonlinear variables.
In the MultiApp you can do anything you want, a solve, use auxvariables, parsedAux etc

Guillaume

**Reply by GiudGiud** [reply]
I saw your other post
CoupledValueFunctionIC seems to be a workable solution too. I m posting it here for the future onlooker
https://mooseframework.inl.gov/source/ics/CoupledValueFunctionIC.html
